### PR TITLE
#11740: Extend post commit coverage and add sweep test

### PR DESCRIPTION
**Pull request by Aswinmcw**
Tracking Issue : #11740 

Improve coverage in existing unit test and add new sweep test for N300

### Checklist
- [x] Post commit CI passes - https://github.com/tenstorrent/tt-metal/actions/runs/11011118826
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: tests/sweep_framework/sweeps/ccl/all_gather_n300.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        "num_links": [1],
+        "input_shape": input_shapes,
+        "dim": [0, 1, 2, 3],
+        "layout": [ttnn.ROW_MAJOR_LAYOUT, ttnn.TILE_LAYOUT],
+        "input_dtype": [ttnn.bfloat16, ttnn.bfloat8_b],
+        "mem_config": [
+            ttnn.MemoryConfig(buffer_type=ttnn.BufferType.DRAM),
+            ttnn.MemoryConfig(buffer_type=ttnn.BufferType.L1),
+        ],
+        "enable_async": [True, False],
```
Comment (SeanNijjar) [inline tests/sweep_framework/sweeps/ccl/all_gather_n300.py:48]: for these sweeps we should only need to run in async mode. I don't think I mentioned this before (will also cut number of cases in half)

**Review by SeanNijjar — APPROVED** [review]
Please just ensure these tests are running properly (though don't necessarily need to pass)

File: tests/sweep_framework/sweeps/ccl/all_gather_n300.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+def mesh_device_fixture():
+    assert ttnn.get_num_devices() == 2, "Not N300!"
+
+    num_devices = ttnn.GetNumAvailableDevices()
+    device_ids = [i for i in range(num_devices)]
+
+    devices = ttnn.CreateDevices(device_ids)
+
+    yield ([devices[i] for i in range(num_devices)], "N300 Fixture")
+
```
Comment (jdesousa-TT) [inline tests/sweep_framework/sweeps/ccl/all_gather_n300.py:78]: Is `del devices` going to properly close the devices? I think you should loop through and call `ttnn.close_device()` on each here

**Review by jdesousa-TT — CHANGES_REQUESTED** [review]

File: tests/sweep_framework/sweeps/ccl/all_gather_n300.py (hunk 2 above)
Comment (Aswinmcw) [reply tests/sweep_framework/sweeps/ccl/all_gather_n300.py:78]: @jdesousa-TT , i have tried looping through devices and closing but it seems like `ttnn.close_device(devices[0])` itself closes both 1 and 2 chip 
Screenshot 2024-09-27 at 7 06 38 PM

Comment (jdesousa-TT) [reply tests/sweep_framework/sweeps/ccl/all_gather_n300.py:78]: That's fine, as long as they get closed properly. Thanks

**Review by jdesousa-TT — APPROVED** [review]

**Review by eyonland — APPROVED** [review]

**Review by TT-BrianLiu — APPROVED** [review]

**Pull request merged**